components/formats-gpl/src/loci/formats/in/ZeissCZIReader.java

```diff
@@ -3400,1201 +4146,481 @@ private void translateExperiment(Element root) throws FormatException {
-    byte a = stream.readByte();
-    // if high bit set, read next byte
-    // at most 3 bytes read
-    if ((a & 0x80) == 0x80) {
-      byte b = stream.readByte();
-      if ((b & 0x80) == 0x80) {
-        byte c = stream.readByte();
-        return (c << 14) | ((b & 0x7f) << 7) | (a & 0x7f);
-      }
-      return (b << 7) | (a & 0x7f);
+    static class Channel {
+      public String name;
+      public String color;
+      public IlluminationType illumination;
+      public AcquisitionMode acquisitionMode;
+      public String emission;
+      public String excitation;
+      public String pinhole;
+      public Double exposure;
+      public Double gain;
+      public String fluor;
+      public String filterSetRef;
     }
-    return a & 0xff;
-  }
 
-  /** Segment with ID "ZISRAWDIRECTORY". */
-  class Directory extends Segment {
-    public DirectoryEntry[] entries;
+    private static class SceneProperties {
+      final List<XYZLength> pos = new ArrayList<>();
+      String name;
 
-    @Override
-    public void fillInData() throws IOException {
-      super.fillInData();
-
-      RandomAccessInputStream s = getStream();
-      try {
-        s.order(isLittleEndian());
-        s.seek(startingPosition + HEADER_SIZE);
-
-        int entryCount = s.readInt();
-        s.skipBytes(124);
-        entries = new DirectoryEntry[entryCount];
-        for (int i=0; i<entryCount; i++) {
-          entries[i] = new DirectoryEntry(s);
-        }
-      }
-      finally {
-        if (stream == null) {
-          s.close();
+      public Length getMinPosXInMicrons() {
+        if (pos.size()==0) {
+          return null;//new Length(0, UNITS.MICROMETER);
         }
-      }
-    }
-  }
-
-  /** Segment with ID "ZISRAWATTDIR". */
-  class AttachmentDirectory extends Segment {
-    public AttachmentEntry[] entries;
-
-    @Override
-    public void fillInData() throws IOException {
-      super.fillInData();
-
-      RandomAccessInputStream s = getStream();
-      try {
-        s.order(isLittleEndian());
-        s.seek(startingPosition + HEADER_SIZE);
-
-        int entryCount = s.readInt();
-        s.skipBytes(252);
-        entries = new AttachmentEntry[entryCount];
-        for (int i=0; i<entryCount; i++) {
-          entries[i] = new AttachmentEntry(s);
+        double minX = Double.MAX_VALUE;
+        for (XYZLength positionLocation : pos) {
+          if ((positionLocation.pX==null)||(positionLocation.pX.value()==null)) {
+            return null;
+          }
+          if (positionLocation.pX.value().doubleValue()<minX) {
+            minX = positionLocation.pX.value().doubleValue();
+          }
         }
+        return new Length(minX, pos.get(0).pX.unit());
       }
-      finally {
-        if (stream == null) {
-          s.close();
+      public Length getMinPosYInMicrons() {
+        if (pos.size()==0) {
+          return null;//new Length(0, UNITS.MICROMETER);
```

Comment:
is this comment deliberately left in place or the null is for debugging?

Reply:
Oh!! Good catch... Maybe that's a reason why we get some NPE in the test files. But I'm not sure this case should ever happen. Logging an error would probably be the first thing to test